modules/apps/transfer/keeper/relay.go

```diff
@@ -220,35 +220,26 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 			// sender chain is not the source, unescrow tokens
 
 			// remove prefix added by sender chain
-			voucherPrefix := types.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
-			unprefixedDenom := fullDenomPath[len(voucherPrefix):]
```

Comment:
only made necessary changes in this file, future pr's will cleanup the flow more